gnosis/safe-react
# refactor: cleanup `create`/`processTransaction`

## What it solves
Unconvential `TxSender` instantiation, cleaner types and external state mutation. 

## How this PR fixes it
`TxSender` has been converted to use a `constructor()` instead of the unconventional `prepare()` method. The state is also no longer mutated from outside.

`this` is never modified apart from inside the `constructor()`. All data the `TxSender` requires is built in the respective action first, assigning default values where necessary, e.g. in `txArgs`.

## How to test it
This needs a full regression testing of transaction creation/processing.

- Create and queue, create and immediately execute, confirm, confirm and execute transactions across different (hardware and smart contract) wallets as (non-)owners.
